contracts/order/mixins/ArtOrderFundsDistributor.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+contract ArtOrderFundsDistributor is Initializable {
+    address public feeRecipient;
+    uint256 public feeAmount;
+
+    error FundsTransferFailed();
+    error InvalidEtherAmount();
+
+    function __ArtOrderFundsDistributor_init(
+        address _feeRecipientAddress,
+        uint256 _feeAmount
+    ) internal onlyInitializing {
+        _updateDistributorConfiguration(_feeRecipientAddress, _feeAmount);
+    }
+
+    function _lockOrderFunds(
+        address orderer,
+        uint256 price,
+        address paymentErc20TokenAddress
+    ) internal {
+        if (paymentErc20TokenAddress == address(0)) {
+            if (msg.value != price) revert InvalidEtherAmount();
+        } else {
+            if (msg.value != 0) revert InvalidEtherAmount();
+            IERC20(paymentErc20TokenAddress).transferFrom(
```

Comment:
tranferFrom ti vraca bool. Mozemo to handlat s nase strane i bacit error ako je false